packages/nexrender-worker/src/index.js

```diff
@@ -54,6 +58,11 @@ const start = async (host, secret, settings) => {
     settings = init(Object.assign({}, settings, {
         logger: console,
     }))
+    // onReadyForPickup will be called before each pickup attempt
+    // return false to skip pickup. Return true to pickup
+    if (!settings.onReadyForPickup) {
```

Comment:
Looks good. Only suggestion is rather than mutating settings here, change `nextJob` to check if `settings.onReadyForPickup` is a function and call it if so, otherwise default to true.